<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-07-22" />

<title>Gene Set Analysis</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drug-AUC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kitadalvi/drug-AUC-analysis.Rproj">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene Set Analysis</h1>
<h4 class="date">2024-07-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>drug-AUC-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240628code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240628)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240628code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240628)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkitadalvidrugAUCanalysistree73fb4f5a994194f99b95ff8232c8c74e304b09a3targetblank73fb4f5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/tree/73fb4f5a994194f99b95ff8232c8c74e304b09a3" target="_blank">73fb4f5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkitadalvidrugAUCanalysistree73fb4f5a994194f99b95ff8232c8c74e304b09a3targetblank73fb4f5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kitadalvi/drug-AUC-analysis/tree/73fb4f5a994194f99b95ff8232c8c74e304b09a3" target="_blank">73fb4f5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/gene_set_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/gene_set_analysis.Rmd</code>) and
HTML (<code>docs/gene_set_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/analysis/gene_set_analysis.Rmd" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
<td>
added GSEA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kitadalvi/drug-AUC-analysis/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/gene_set_analysis.html" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
<td>
added GSEA
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overviewnotes" class="section level2">
<h2>Overview/Notes</h2>
<p>Plots used for analysis: • Dot plot:number of genes belonging to each
GO-term and associated adjusted p-values</p>
<p>• Enrichment map: Shows overlap between genes in different GO
terms</p>
<p>• Category Net plot:Linkages of genes and biological concepts as a
network</p>
<p>• Ridge Plot: Density pltos of expression for different gene sets to
determine up/downregulated pathways</p>
</div>
<div
id="analysis-of-significantly-correlated-genes-across-top1-inhibitors"
class="section level2">
<h2>Analysis of significantly correlated genes across TOP1
inhibitors</h2>
<div id="significant-genes" class="section level3">
<h3>Significant genes</h3>
<pre class="r"><code>#loading in data
top1_sig_genes &lt;- read.delim(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/top1_drugs_genes.txt&#39;, header=T)
head(top1_sig_genes)</code></pre>
<pre><code>       genes      logFC    logCPM       LR       PValue          FDR
1      ZFPM2  0.3087779 1.8306206 93.88924 3.336836e-22 5.409345e-18
2      IGFL2 -0.1536392 1.3589416 84.83488 3.243520e-20 2.629035e-16
3 AL157931.2  0.3432817 5.6722277 64.93880 7.726072e-16 4.174912e-12
4 AC022596.1  0.1622893 1.3221242 54.86168 1.293179e-13 5.240931e-10
5     NUS1P2  0.3051329 5.2650851 51.62835 6.706618e-13 2.133393e-09
6      PALMD  0.2111364 0.3114543 51.30781 7.896094e-13 2.133393e-09</code></pre>
</div>
<div id="preparing-input" class="section level3">
<h3>Preparing input</h3>
<pre class="r"><code>gene_list &lt;- top1_sig_genes$logFC
names(gene_list)&lt;- top1_sig_genes$genes
gene_list=sort(gene_list, decreasing = TRUE)

top1_gse &lt;- gseGO(geneList=gene_list, 
             ont =&quot;ALL&quot;, 
             keyType = &quot;ALIAS&quot;, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = &quot;none&quot;)</code></pre>
</div>
<div id="dotplot" class="section level3">
<h3>DotPlot</h3>
<pre class="r"><code>require(DOSE)
dotplot(top1_gse, showCategory=8,split=&quot;.sign&quot;,label_format=50,color=&#39;p.adjust&#39;)+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle(&quot;dotplot for TOP1 inhibitors&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top1_dotplot-1">
Past versions of top1_dotplot-1.png
</button>
</p>
<div id="fig-top1_dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/top1_dotplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-map" class="section level3">
<h3>Enrichment Map</h3>
<pre class="r"><code>x&lt;- pairwise_termsim(top1_gse)
emapplot(x,layout.params=list(layout=&#39;circle&#39;))+
  ggtitle(&quot;enrichment map for TOP1 inhibitors&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_enrichment_map-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top1_enrichment_map-1">
Past versions of top1_enrichment_map-1.png
</button>
</p>
<div id="fig-top1_enrichment_map-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/top1_enrichment_map-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="category-netplot" class="section level3">
<h3>Category Netplot</h3>
<pre class="r"><code>cnetplot(top1_gse, categorySize=&quot;pvalue&quot;, color.params=list(foldChange=gene_list,category=&#39;deeppink1&#39;))+
  ggtitle(&quot;By p-value&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_cnetPlot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top1_cnetPlot-1">
Past versions of top1_cnetPlot-1.png
</button>
</p>
<div id="fig-top1_cnetPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/top1_cnetPlot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(top1_gse, node_label=&quot;all&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By both gene category and name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_cnetPlot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(top1_gse, node_label=&quot;category&quot;, showCategory=10, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;))+
  ggtitle(&quot;By gene category&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_cnetPlot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(top1_gse, node_label=&quot;gene&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By gene name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1_cnetPlot-4.png" width="672" style="display: block; margin: auto;" />
### Ridgeplot</p>
<pre class="r"><code>ridgeplot(top1_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle(&quot;ridgeplot for TOP1 inhibitors&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/top1__ridgeplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top1__ridgeplot-1">
Past versions of top1__ridgeplot-1.png
</button>
</p>
<div id="fig-top1__ridgeplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/top1__ridgeplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="analysis-of-significantly-correlated-genes-across-platinum-compounds"
class="section level2">
<h2>Analysis of significantly correlated genes across platinum
compounds</h2>
<div id="significant-genes-1" class="section level3">
<h3>Significant genes</h3>
<pre class="r"><code>#loading in data
plat_sig_genes &lt;- read.delim(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/plat_drugs_genes.txt&#39;, header=T)
head(plat_sig_genes)</code></pre>
<pre><code>    genes      logFC    logCPM        LR       PValue          FDR
1  CCDC43 0.02505341  5.762231 191.66636 1.375972e-43 2.230589e-39
2 GPATCH8 0.02323835  8.024142 171.65390 3.220680e-39 2.610522e-35
3    ACLY 0.02568084 10.161320 131.18368 2.257231e-30 1.219732e-26
4 RASL11A 0.03333610  3.321720 120.21007 5.690421e-28 2.306185e-24
5 C5orf17 0.04955097  2.232800  97.14530 6.442195e-23 2.088688e-19
6  EFTUD2 0.01855204  7.721094  85.32133 2.536121e-20 6.852175e-17</code></pre>
</div>
<div id="preparing-input-1" class="section level3">
<h3>Preparing input</h3>
<pre class="r"><code>gene_list &lt;- plat_sig_genes$logFC
names(gene_list)&lt;- plat_sig_genes$genes
gene_list=sort(gene_list, decreasing = TRUE)

gse &lt;- gseGO(geneList=gene_list, 
             ont =&quot;ALL&quot;, 
             keyType = &quot;ALIAS&quot;, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = &quot;none&quot;)</code></pre>
</div>
<div id="dotplot-1" class="section level3">
<h3>DotPlot</h3>
<pre class="r"><code>require(DOSE)
dotplot(gse, showCategory=8, split=&quot;.sign&quot;,label_format=50,color=&#39;p.adjust&#39;)+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle(&quot;dotplot for platinum based compounds&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-platinum_compounds_dotplot-1">
Past versions of platinum_compounds_dotplot-1.png
</button>
</p>
<div id="fig-platinum_compounds_dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/platinum_compounds_dotplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-map-1" class="section level3">
<h3>Enrichment Map</h3>
<pre class="r"><code>x&lt;- pairwise_termsim(gse)
emapplot(x, showCategory=40)+
  ggtitle(&quot;enrichment map for platinum based compounds&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_enrichment_map-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-platinum_compounds_enrichment_map-1">
Past versions of platinum_compounds_enrichment_map-1.png
</button>
</p>
<div id="fig-platinum_compounds_enrichment_map-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/platinum_compounds_enrichment_map-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="category-netplot-1" class="section level3">
<h3>Category Netplot</h3>
<pre class="r"><code>cnetplot(gse, categorySize=&quot;pvalue&quot;, color.params=list(foldChange=gene_list,category=&#39;deeppink1&#39;))+
  ggtitle(&quot;By p-value&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_cnetPlot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-platinum_compounds_cnetPlot-1">
Past versions of platinum_compounds_cnetPlot-1.png
</button>
</p>
<div id="fig-platinum_compounds_cnetPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/platinum_compounds_cnetPlot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(gse, node_label=&quot;all&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By both gene category and name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_cnetPlot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(gse, node_label=&quot;category&quot;, showCategory=10, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;))+
  ggtitle(&quot;By gene category&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_cnetPlot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(gse, node_label=&quot;gene&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By gene name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_cnetPlot-4.png" width="672" style="display: block; margin: auto;" />
### Ridgeplot</p>
<pre class="r"><code>ridgeplot(gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle(&quot;ridgeplot for platinum based compounds&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/platinum_compounds_ridgeplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-platinum_compounds_ridgeplot-1">
Past versions of platinum_compounds_ridgeplot-1.png
</button>
</p>
<div id="fig-platinum_compounds_ridgeplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/platinum_compounds_ridgeplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="analysis-of-significantly-correlated-genes-across-cytotoxic-antibiotics"
class="section level2">
<h2>Analysis of significantly correlated genes across cytotoxic
antibiotics</h2>
<div id="significant-genes-2" class="section level3">
<h3>Significant genes</h3>
<pre class="r"><code>#loading in data
cytotoxic_antibiotic_genes &lt;- read.delim(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/cytotoxic_antibiotic_genes.txt&#39;, header=T)
head(cytotoxic_antibiotic_genes)</code></pre>
<pre><code>    genes       logFC   logCPM       LR       PValue          FDR
1   IGFL2 -0.11638406 1.359505 65.23729 6.640095e-16 1.076426e-11
2   MYO3A  0.11501729 1.058253 57.43266 3.497516e-14 2.834911e-10
3  LRRTM1  0.07617468 3.212449 39.62463 3.077765e-10 1.663122e-06
4    LHFP  0.08038259 0.865455 38.56613 5.292950e-10 2.145100e-06
5 SNORD3A -0.05365899 4.851899 31.96733 1.567876e-08 5.083367e-05
6    ISL1  0.04426481 2.973359 30.65544 3.081615e-08 8.326010e-05</code></pre>
</div>
<div id="preparing-input-2" class="section level3">
<h3>Preparing input</h3>
<pre class="r"><code>ca_gene_list &lt;- cytotoxic_antibiotic_genes$logFC
names(ca_gene_list)&lt;- cytotoxic_antibiotic_genes$genes
ca_gene_list=sort(ca_gene_list, decreasing = TRUE)

ca_gse &lt;- gseGO(geneList=ca_gene_list, 
             ont =&quot;ALL&quot;, 
             keyType = &quot;ALIAS&quot;, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = &quot;none&quot;)</code></pre>
</div>
<div id="dotplot-2" class="section level3">
<h3>DotPlot</h3>
<pre class="r"><code>require(DOSE)
dotplot(ca_gse, showCategory=8, split=&quot;.sign&quot;,label_format=50,color=&#39;p.adjust&#39;)+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle(&quot;dotplot for cytotoxic antibiotics&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cytotoxic_antibiotics_dotplot-1">
Past versions of cytotoxic_antibiotics_dotplot-1.png
</button>
</p>
<div id="fig-cytotoxic_antibiotics_dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_dotplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-map-2" class="section level3">
<h3>Enrichment Map</h3>
<pre class="r"><code>x&lt;- pairwise_termsim(ca_gse)
emapplot(x, showCategory=40)+
  ggtitle(&quot;enrichment map for cytotoxic antibiotics&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_enrichment_map-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cytotoxic_antibiotics_enrichment_map-1">
Past versions of cytotoxic_antibiotics_enrichment_map-1.png
</button>
</p>
<div id="fig-cytotoxic_antibiotics_enrichment_map-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_enrichment_map-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="category-netplot-2" class="section level3">
<h3>Category Netplot</h3>
<pre class="r"><code>cnetplot(ca_gse, categorySize=&quot;pvalue&quot;, color.params=list(foldChange=gene_list,category=&#39;deeppink1&#39;))+
  ggtitle(&quot;By p-value&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_cnetPlot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cytotoxic_antibiotics_cnetPlot-1">
Past versions of cytotoxic_antibiotics_cnetPlot-1.png
</button>
</p>
<div id="fig-cytotoxic_antibiotics_cnetPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_cnetPlot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(ca_gse, node_label=&quot;all&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By both gene category and name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_cnetPlot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(ca_gse, node_label=&quot;category&quot;, showCategory=10, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;))+
  ggtitle(&quot;By gene category&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_cnetPlot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(ca_gse, node_label=&quot;gene&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By gene name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_cnetPlot-4.png" width="672" style="display: block; margin: auto;" />
### Ridgeplot</p>
<pre class="r"><code>ridgeplot(ca_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle(&quot;ridgeplot for cytotoxic antibiotics&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_ridgeplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cytotoxic_antibiotics_ridgeplot-1">
Past versions of cytotoxic_antibiotics_ridgeplot-1.png
</button>
</p>
<div id="fig-cytotoxic_antibiotics_ridgeplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/cytotoxic_antibiotics_ridgeplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="analysis-of-significantly-correlated-genes-across-taxanes"
class="section level2">
<h2>Analysis of significantly correlated genes across taxanes</h2>
<div id="significant-genes-3" class="section level3">
<h3>Significant genes</h3>
<pre class="r"><code>#loading in data
taxane_genes &lt;- read.delim(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/taxane_genes.txt&#39;, header=T)
head(taxane_genes)</code></pre>
<pre><code>       genes      logFC    logCPM       LR       PValue          FDR
1      ZFPM2  0.2675151 1.8304000 88.93729 4.075282e-21 6.606440e-17
2      ENPEP -0.1855867 0.4347222 78.90367 6.521340e-19 5.285872e-15
3 AC022389.1  0.2022841 2.2686059 61.91165 3.592185e-15 1.941097e-11
4 AC022596.1  0.1349096 1.3218233 53.27907 2.893726e-13 1.172755e-09
5      CCDC8  0.1425379 2.2724972 48.38518 3.502048e-12 1.135434e-08
6    COL12A1  0.1200130 3.7930005 44.98151 1.989039e-11 5.374051e-08</code></pre>
</div>
<div id="preparing-input-3" class="section level3">
<h3>Preparing input</h3>
<pre class="r"><code>tax_gene_list &lt;- taxane_genes$logFC
names(tax_gene_list)&lt;- taxane_genes$genes
tax_gene_list=sort(tax_gene_list, decreasing = TRUE)

tax_gse &lt;- gseGO(geneList=tax_gene_list, 
             ont =&quot;ALL&quot;, 
             keyType = &quot;ALIAS&quot;, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = &quot;none&quot;)</code></pre>
</div>
<div id="dotplot-3" class="section level3">
<h3>DotPlot</h3>
<pre class="r"><code>require(DOSE)
dotplot(tax_gse, showCategory=8, split=&quot;.sign&quot;,label_format=50,color=&#39;p.adjust&#39;)+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle(&quot;dotplot for taxanes&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-taxanes_dotplot-1">
Past versions of taxanes_dotplot-1.png
</button>
</p>
<div id="fig-taxanes_dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/taxanes_dotplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-map-3" class="section level3">
<h3>Enrichment Map</h3>
<pre class="r"><code>x&lt;- pairwise_termsim(tax_gse)
emapplot(x, showCategory=40)+
  ggtitle(&quot;enrichment map for taxanes&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_enrichment_map-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-taxanes_enrichment_map-1">
Past versions of taxanes_enrichment_map-1.png
</button>
</p>
<div id="fig-taxanes_enrichment_map-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/taxanes_enrichment_map-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="category-netplot-3" class="section level3">
<h3>Category Netplot</h3>
<pre class="r"><code>cnetplot(tax_gse, categorySize=&quot;pvalue&quot;, color.params=list(foldChange=gene_list,category=&#39;deeppink1&#39;))+
  ggtitle(&quot;By p-value&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_cnetPlot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-taxanes_cnetPlot-1">
Past versions of taxanes_cnetPlot-1.png
</button>
</p>
<div id="fig-taxanes_cnetPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/taxanes_cnetPlot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(tax_gse, node_label=&quot;all&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By both gene category and name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_cnetPlot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(tax_gse, node_label=&quot;category&quot;, showCategory=10, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;))+
  ggtitle(&quot;By gene category&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_cnetPlot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cnetplot(tax_gse, node_label=&quot;gene&quot;, color.params=list(category=&#39;deeppink1&#39;, gene=&#39;darkmagenta&#39;), showCategory=10)+
  ggtitle(&quot;By gene name&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_cnetPlot-4.png" width="672" style="display: block; margin: auto;" />
### Ridgeplot</p>
<pre class="r"><code>ridgeplot(tax_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle(&quot;ridgeplot for taxanes&quot;)</code></pre>
<p><img src="figure/gene_set_analysis.Rmd/taxanes_ridgeplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-taxanes_ridgeplot-1">
Past versions of taxanes_ridgeplot-1.png
</button>
</p>
<div id="fig-taxanes_ridgeplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/73fb4f5a994194f99b95ff8232c8c74e304b09a3/docs/figure/gene_set_analysis.Rmd/taxanes_ridgeplot-1.png" target="_blank">73fb4f5</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] DOSE_3.31.2            ggpubr_0.6.0           ggplot2_3.5.1         
 [4] org.Hs.eg.db_3.19.1    AnnotationDbi_1.67.0   IRanges_2.39.2        
 [7] S4Vectors_0.43.2       Biobase_2.65.0         BiocGenerics_0.51.0   
[10] enrichplot_1.25.0      clusterProfiler_4.13.0 edgeR_4.3.5           
[13] limma_3.61.4           workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.8         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.27         
  [7] fs_1.6.4                zlibbioc_1.51.1         vctrs_0.6.5            
 [10] memoise_2.0.1           ggtree_3.13.0           rstatix_0.7.2          
 [13] htmltools_0.5.8.1       broom_1.0.6             gridGraphics_0.5-1     
 [16] sass_0.4.9              bslib_0.7.0             plyr_1.8.9             
 [19] cachem_1.1.0            whisker_0.4.1           igraph_2.0.3           
 [22] lifecycle_1.0.4         pkgconfig_2.0.3         gson_0.1.0             
 [25] Matrix_1.7-0            R6_2.5.1                fastmap_1.2.0          
 [28] GenomeInfoDbData_1.2.12 digest_0.6.36           aplot_0.2.3            
 [31] ggnewscale_0.5.0        colorspace_2.1-0        patchwork_1.2.0        
 [34] ps_1.7.7                rprojroot_2.0.4         RSQLite_2.3.7          
 [37] labeling_0.4.3          fansi_1.0.6             abind_1.4-5            
 [40] httr_1.4.7              polyclip_1.10-6         compiler_4.4.0         
 [43] bit64_4.0.5             withr_3.0.0             backports_1.5.0        
 [46] BiocParallel_1.39.0     carData_3.0-5           viridis_0.6.5          
 [49] DBI_1.2.3               highr_0.11              ggforce_0.4.2          
 [52] ggsignif_0.6.4          MASS_7.3-61             HDO.db_0.99.1          
 [55] tools_4.4.0             scatterpie_0.2.3        ape_5.8                
 [58] httpuv_1.6.15           glue_1.7.0              callr_3.7.6            
 [61] nlme_3.1-165            GOSemSim_2.31.0         promises_1.3.0         
 [64] shadowtext_0.1.4        grid_4.4.0              getPass_0.2-4          
 [67] reshape2_1.4.4          fgsea_1.31.0            generics_0.1.3         
 [70] gtable_0.3.5            tidyr_1.3.1             data.table_1.15.4      
 [73] car_3.1-2               tidygraph_1.3.1         utf8_1.2.4             
 [76] XVector_0.45.0          ggrepel_0.9.5           pillar_1.9.0           
 [79] stringr_1.5.1           yulab.utils_0.1.4       later_1.3.2            
 [82] splines_4.4.0           dplyr_1.1.4             tweenr_2.0.3           
 [85] treeio_1.29.0           lattice_0.22-6          bit_4.0.5              
 [88] tidyselect_1.2.1        GO.db_3.19.1            locfit_1.5-9.10        
 [91] Biostrings_2.73.1       knitr_1.48              git2r_0.33.0           
 [94] gridExtra_2.3           xfun_0.46               graphlayouts_1.1.1     
 [97] statmod_1.5.0           stringi_1.8.4           UCSC.utils_1.1.0       
[100] lazyeval_0.2.2          ggfun_0.1.5             yaml_2.3.9             
[103] evaluate_0.24.0         codetools_0.2-20        ggraph_2.2.1           
[106] tibble_3.2.1            qvalue_2.37.0           ggplotify_0.1.2        
[109] cli_3.6.3               munsell_0.5.1           processx_3.8.4         
[112] jquerylib_0.1.4         Rcpp_1.0.13             GenomeInfoDb_1.41.1    
[115] png_0.1-8               parallel_4.4.0          blob_1.2.4             
[118] viridisLite_0.4.2       tidytree_0.4.6          ggridges_0.5.6         
[121] scales_1.3.0            purrr_1.0.2             crayon_1.5.3           
[124] rlang_1.1.4             cowplot_1.1.3           fastmatch_1.1-4        
[127] KEGGREST_1.45.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
