<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-08-27" />

<title>Drug AUC Model</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">drug-AUC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="MDSplots.html">QC</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_Analysis.html">RNA-seq Analysis</a>
    </li>
    <li>
      <a href="Updated_DrugAUC.html">GLM Analysis</a>
    </li>
    <li>
      <a href="gene_set_analysis.html">Gene Set Enrichment</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kitadalvi/drug-AUC-analysis.Rproj">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Drug AUC Model</h1>
<h4 class="date">2024-08-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>drug-AUC-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240628code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240628)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240628code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240628)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkitadalvidrugAUCanalysistree335a7e92ffad0eefac6d44f35d440068f5c3e1betargetblank335a7e9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/tree/335a7e92ffad0eefac6d44f35d440068f5c3e1be" target="_blank">335a7e9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkitadalvidrugAUCanalysistree335a7e92ffad0eefac6d44f35d440068f5c3e1betargetblank335a7e9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kitadalvi/drug-AUC-analysis/tree/335a7e92ffad0eefac6d44f35d440068f5c3e1be" target="_blank">335a7e9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Unstaged changes:
    Modified:   analysis/Updated_DrugAUC.Rmd
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Updated_DrugAUC.Rmd</code>) and
HTML (<code>docs/Updated_DrugAUC.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/335a7e92ffad0eefac6d44f35d440068f5c3e1be/analysis/Updated_DrugAUC.Rmd" target="_blank">335a7e9</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-25
</td>
<td>
RNA-seq + refitted linear model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kitadalvi/drug-AUC-analysis/335a7e92ffad0eefac6d44f35d440068f5c3e1be/docs/Updated_DrugAUC.html" target="_blank">335a7e9</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-25
</td>
<td>
RNA-seq + refitted linear model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kitadalvi/drug-AUC-analysis/a500793f4eb87bbf1438860ceba5c4b26a239fea/docs/Updated_DrugAUC.html" target="_blank">a500793</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
<td>
Update Updated_DrugAUC.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kitadalvi/drug-AUC-analysis/96e488c8e060c011d4a4cc360b59b9513534c144/docs/Updated_DrugAUC.html" target="_blank">96e488c</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
<td>
Linked corr analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/analysis/Updated_DrugAUC.Rmd" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
<td>
updated AUC analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kitadalvi/drug-AUC-analysis/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/Updated_DrugAUC.html" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
<td>
updated AUC analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="rna-seq-data" class="section level2">
<h2>RNA-seq data</h2>
<p>A quick look at the data:</p>
<pre class="r"><code># loading in data
rawdata &lt;- read.delim2(&#39;/Users/dalvinikita/Documents/RNA-seq Analysis/Sequencing Data/collated.count.tsv&#39;, header = TRUE, check.names = FALSE, stringsAsFactors = FALSE)

head(rawdata) </code></pre>
<pre><code>           Gene ORG16-1Y ORG16T ORG38H ORG38HL ORG38 ORG38D ORG38L ORG41 ORG44H
1 1   5S_rRNA.1        0      0      1       0     0      0      0     0      0
2 2 5S_rRNA.100        0      0      0       0     0      2      0     0      0
3 3 5S_rRNA.103        0      0      0       0     0      0      0     1      0
4 4 5S_rRNA.105        0      1      0       0     1      1      1     1      0
5 5 5S_rRNA.106        0      0      0       0     0      0      0     0      0
6 6 5S_rRNA.110        0      0      0       0     1      1      1     0      0
  ORG44 ORG46H ORG46 ORG47H ORG47 ORG49 ORG49T ORG56 ORG60 ORG60T ORG62 ORG63T
1     0      0     0      1     0     0      0     0     1      0     0      0
2     0      0     0      0     0     0      0     0     0      0     0      0
3     0      0     0      0     0     0      0     0     0      0     0      0
4     1      0     0      0     0     1      1     0     0      0     1      0
5     1      0     0      0     0     0      0     0     0      0     0      0
6     0      0     0      0     0     0      0     0     0      0     0      0
  ORG66 ORG71 ORG71T ORG73 ORG73R CAF55 CAF57 CAF58 CAF62 CAF63 HOSE-ELF3-KO
1     0     1      0     0      2     0     0     0     0     0            0
2     0     0      0     0      0     0     0     0     0     0            0
3     0     0      0     0      0     0     0     0     0     0            0
4     0     1      0     0      0     1     0     0     0     0            0
5     0     0      0     0      0     0     0     0     0     0            0
6     0     0      0     0      0     0     1     0     0     0            0
  HOSE JHOM-ELF3-KO JHOM MCAS-ELF3-KO MCAS RMUG-ELF3-KO RMUG SLC58 VOA4627 BJ
1    1            0    0            0    0            0    0     0       0  0
2    0            0    0            0    0            0    0     0       0  0
3    0            0    0            0    0            0    0     0       0  0
4    0            0    0            1    0            0    0     0       0  0
5    0            0    0            0    0            0    0     0       0  0
6    0            0    0            0    0            1    0     0       0  0
  HFF
1   0
2   0
3   0
4   0
5   0
6   0</code></pre>
<pre class="r"><code>sampleinfo &lt;- read.delim2(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt&#39;, header=TRUE, check.names = FALSE, stringsAsFactors = TRUE)

head(sampleinfo)</code></pre>
<pre><code>   Sample ID Type PATIENT PATTERN Batch Media Mitomycin Doxorubicin
1 ORG-38-ORG  ORG   ORG38     INF     1   ORG     156.1       141.2
2      ORG41  ORG   ORG41     EXP     1   ORG      80.1       99.32
3      ORG46  ORG   ORG46     EXP     2   ORG     80.49       120.3
4      ORG49  ORG   ORG49     INF     2   ORG     41.72       78.31
5      ORG60  ORG   ORG60     INF     2   ORG     166.7       148.2
6      ORG66  ORG   ORG66     EXP     3   ORG     113.4         132
  Staurosporine Topotecan Irinotecan Oxaliplatin Gemcitabine Carboplatin
1         35.09     118.1      198.4       223.8       155.1       256.8
2         16.83     44.23      165.9       225.3       76.88         250
3         32.21     30.97        117       215.7       51.96       221.5
4         5.847     27.42      129.2       498.9       35.48       300.7
5         45.89     70.09      167.5       211.1       145.7       230.7
6         21.27     34.27      164.8       193.5       80.76       222.8
  Docetaxel Cisplatin Paclitaxel
1     155.2     205.1      157.3
2     101.2     190.2      103.1
3     77.56     187.1      69.78
4     63.52     389.5      62.57
5     119.3     195.2      113.6
6     68.85     193.8      67.05</code></pre>
<div id="qc-and-normalisation" class="section level3">
<h3>QC and Normalisation</h3>
<pre><code>keep
FALSE  TRUE 
21232 17551 </code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-2-1.png" width="2880" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/figure/Updated_DrugAUC.Rmd/unnamed-chunk-2-1.png" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mds-plots" class="section level3">
<h3>MDS plots</h3>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-3-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/figure/Updated_DrugAUC.Rmd/unnamed-chunk-3-1.png" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="fitting-a-glm" class="section level2 tabset">
<h2 class="tabset">Fitting a GLM</h2>
<pre class="r"><code>sampleinfo &lt;- read.delim2(&#39;/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt&#39;, header=TRUE, check.names = FALSE, stringsAsFactors = FALSE)

y&lt;- DGEList(counts=rawdata[,c(7,10,14,17,20,24,25,28)], genes=rawdata[,2])

batch &lt;- as.factor(sampleinfo$Batch)</code></pre>
<div id="mitomycin" class="section level3">
<h3>Mitomycin</h3>
<pre class="r"><code>mito_counts &lt;- y
design &lt;- model.matrix(~batch+sampleinfo$Mitomycin)
mitomycin &lt;- as.numeric(sampleinfo$Mitomycin)
design &lt;- cbind(design[,1:3],mitomycin)
rownames(design)&lt;- sampleinfo$`Sample ID`</code></pre>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 mitomycin
ORG-38-ORG           1      0      0    156.10
ORG41                1      0      0     80.10
ORG46                1      1      0     80.49
ORG49                1      1      0     41.72
ORG60                1      1      0    166.70
ORG66                1      0      1    113.40
ORG71                1      0      1    106.60
ORG73R               1      0      1     75.83</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(mito_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>mito_counts &lt;- mito_counts[keep, , keep.lib.sizes=FALSE]

#normalizing library sizes
mito_counts &lt;- calcNormFactors(mito_counts)

#fit GLMs
fit &lt;- glmQLFit(mito_counts, design)
mito_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;mitomycin&#39;))
genes &lt;- topTags(mito_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-212d6734bb3be6d86890" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-212d6734bb3be6d86890">{"x":{"filter":"none","vertical":false,"data":[["32272","35595","31133","32262","7705","12551","8323","5454","30618","14673"],["TGM2","ZFP57","SNORD3A","TGFBI","C21orf90","FGF19","CASP1","ALDH1A2","SLC9A3","IGFL2"],[0.03469391849395809,-0.192944443104179,-0.04845502934558709,-0.04469347826996968,-0.06276996176603357,0.05046409751559885,-0.0454079524146502,0.06322793797714274,-0.06020011849491749,-0.09691713209188969],[9.302269696324471,2.62193654755737,5.279926279582194,5.074678913630198,1.235091374169878,5.210875081089467,1.375752092843454,3.347807687852173,3.857519217259942,1.801721753043741],[59.81083953462966,57.85770451297351,48.40931991769293,43.85265081620671,29.25704787238555,40.56641986970477,26.41188139431528,26.71944187937722,27.50188893164508,32.2035857566907],[6.675645561631077e-05,0.000109900988284159,0.000138515753554091,0.0001919809059832492,0.0002130964958679626,0.0002461832247453788,0.0002838574831276457,0.0004355463964863594,0.0004523751736612766,0.0004747511061883429],[0.6420450757657163,0.6420450757657163,0.6420450757657163,0.6420450757657163,0.6420450757657163,0.6420450757657163,0.6420450757657163,0.7127401124624085,0.7127401124624085,0.7127401124624085]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- mitomycin

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  mito_counts$cpm &lt;- cpm(mito_counts, log = T) 
  expression_values &lt;- mito_counts$cpm[mito_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-1.png" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-7-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="irinotecan" class="section level3">
<h3>Irinotecan</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 irinotecan
ORG-38-ORG           1      0      0      198.4
ORG41                1      0      0      165.9
ORG46                1      1      0      117.0
ORG49                1      1      0      129.2
ORG60                1      1      0      167.5
ORG66                1      0      1      164.8
ORG71                1      0      1      177.1
ORG73R               1      0      1      151.9</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(irino_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>irino_counts &lt;- irino_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
irino_counts &lt;- calcNormFactors(irino_counts)
#fit GLMs
fit &lt;- glmQLFit(irino_counts, design)
irinotecan_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;irinotecan&#39;))
genes &lt;- topTags(irinotecan_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-6f947d133c2edd746287" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6f947d133c2edd746287">{"x":{"filter":"none","vertical":false,"data":[["32769","17755","11954","14429","13300","7673","16294","9520","25239","11319"],["TNFRSF19","MUC2","EYA4","HS6ST2","GNAS-AS1","C20orf85","LRRTM1","CP","RP11-424M22.3","DSG3"],[0.1370360995469916,0.1663674350498321,0.1793214387537902,0.1177097542252407,0.1027260220334418,0.1786235042549505,0.1568276977120568,0.2029341332577781,0.1281376203423945,-0.160702607741638],[3.93403572579044,6.361799573448365,2.927906564958798,1.461103117493721,0.5511562310834195,3.450442185996617,2.893452902632583,6.933154625774663,1.075638685797932,2.942291681759965],[55.08492334881562,61.7780089806338,45.29574253141676,23.98617423738084,26.13339614242425,33.32652341073127,25.87740116781,28.79511363910509,26.44807598953485,27.36863602495771],[3.377993412643325e-05,7.951981358801949e-05,0.0001719449899278708,0.0003676721841787152,0.0003867732789161978,0.0004108709706362288,0.0004477390482178114,0.0004958771272098213,0.0006238419711630508,0.000693398513501546],[0.5348376970238177,0.6295186042695563,0.9074683418426598,0.9561297518320926,0.9561297518320926,0.9561297518320926,0.9561297518320926,0.9561297518320926,0.9561297518320926,0.9561297518320926]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- irinotecan

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  irino_counts$cpm &lt;- cpm(irino_counts, log = T) 
  expression_values &lt;- irino_counts$cpm[irino_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-1.png" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-10-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="topotecan" class="section level3">
<h3>Topotecan</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 topotecan
ORG-38-ORG           1      0      0    118.10
ORG41                1      0      0     44.23
ORG46                1      1      0     30.97
ORG49                1      1      0     27.42
ORG60                1      1      0     70.09
ORG66                1      0      1     34.27
ORG71                1      0      1     66.50
ORG73R               1      0      1     29.76</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(topotecan_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>topotecan_counts &lt;- topotecan_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
topotecan_counts &lt;- calcNormFactors(topotecan_counts)
#fit GLMs
fit &lt;- glmQLFit(topotecan_counts, design)
topotecan_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;topotecan&#39;))
genes &lt;- topTags(topotecan_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-3a639866ad5cd7a83a56" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3a639866ad5cd7a83a56">{"x":{"filter":"none","vertical":false,"data":[["4969","7922","19190","18706","6606","5616","8476","35604","31246","7673"],["AL157931.2","C6orf103","PALMD","NUS1P2","ATRNL1","ANKRD18B","CCDC37","ZFPM2","SNTN","C20orf85"],[0.2846405313045106,0.1329280042314295,0.1541555418639583,0.2678904768126502,0.1133772419885804,0.07666756911129521,0.1833669928025131,0.176222115036554,0.1205619190210194,0.1595634199544627],[6.234676718086417,1.133927608669762,0.5983068304488153,5.823123107809365,0.3832695284381713,-0.02672047603908564,2.19856484218001,1.657004902026875,0.2447031968410022,3.450442185996617],[194.60091209155,59.79898956517816,61.60793832914239,181.6439593737544,48.44018140285453,35.19612593463823,38.46356140253875,34.09129423000901,31.90512240709892,51.12261796268839],[1.51291495033676e-05,1.740032771881638e-05,1.871119707754521e-05,2.003297388850732e-05,3.279775253180502e-05,0.0001016844003107398,0.0001064037828111772,0.0001400543338181198,0.0001505879440658385,0.0001567907772931989],[0.07929551889418412,0.07929551889418412,0.07929551889418412,0.07929551889418412,0.1038573631672138,0.2406701561784812,0.2406701561784812,0.2432198190848651,0.2432198190848651,0.2432198190848651]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- topotecan

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  topotecan_counts$cpm &lt;- cpm(topotecan_counts, log = T) 
  expression_values &lt;- topotecan_counts$cpm[topotecan_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kitadalvi/drug-AUC-analysis/blob/67e77120ce7816fc4bb9736c330177f9a43084d7/docs/figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-1.png" target="_blank">67e7712</a>
</td>
<td>
kitadalvi
</td>
<td>
2024-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-13-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="doxorubicin" class="section level3">
<h3>Doxorubicin</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 doxorubicin
ORG-38-ORG           1      0      0      141.20
ORG41                1      0      0       99.32
ORG46                1      1      0      120.30
ORG49                1      1      0       78.31
ORG60                1      1      0      148.20
ORG66                1      0      1      132.00
ORG71                1      0      1       88.41
ORG73R               1      0      1      102.60</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(doxorubicin_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>doxorubicin_counts &lt;- doxorubicin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
doxorubicin_counts &lt;- calcNormFactors(doxorubicin_counts)
#fit GLMs
fit &lt;- glmQLFit(doxorubicin_counts, design)
doxorubicin_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;doxorubicin&#39;))
genes &lt;- topTags(doxorubicin_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-005f23e0505edfa78356" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-005f23e0505edfa78356">{"x":{"filter":"none","vertical":false,"data":[["34622","7329","33294","17408","5454","10774","31718","3114","14543","30229"],["VTCN1","C16orf5","TUBB6","MOCS1","ALDH1A2","DAPK2","STOM","AC092296.1","HUNK","SIRPA"],[-0.1138366008271989,-0.07146577944178939,-0.06573605403222299,-0.1122952328228198,0.1012413121180073,-0.07568454983077488,-0.06375622165530323,-0.05499271865163383,-0.1018726113561165,-0.09610164968816311],[2.657101153738732,3.614503062775523,4.58604723755894,2.886480876691061,3.347807687852173,0.5914784588865697,4.779681715778967,2.125444314255893,2.761697383229444,2.936523217555776],[44.05808445653671,60.30682147180229,54.42979864813803,47.93332858838501,30.55380183844287,25.19899186098413,35.42589175437637,28.93510045811906,29.55903854187721,32.22069604419513],[4.120417682698217e-05,6.846362263347769e-05,9.760210465094754e-05,0.0001153921406925342,0.0003387419679728417,0.0003555064624083874,0.000404681408270564,0.0004649659169578337,0.0004856931961590655,0.0005184484888055838],[0.4567509408962233,0.4567509408962233,0.4567509408962233,0.4567509408962233,0.7039705162107633,0.7039705162107633,0.7039705162107633,0.7039705162107633,0.7039705162107633,0.7039705162107633]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- doxorubicin

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  doxorubicin_counts$cpm &lt;- cpm(doxorubicin_counts, log = T) 
  expression_values &lt;- doxorubicin_counts$cpm[doxorubicin_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-16-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="carboplatin" class="section level3">
<h3>Carboplatin</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 carboplatin
ORG-38-ORG           1      0      0       256.8
ORG41                1      0      0       250.0
ORG46                1      1      0       221.5
ORG49                1      1      0       300.7
ORG60                1      1      0       230.7
ORG66                1      0      1       222.8
ORG71                1      0      1       245.3
ORG73R               1      0      1       229.9</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(carboplatin_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>carboplatin_counts &lt;- carboplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
carboplatin_counts &lt;- calcNormFactors(carboplatin_counts)
#fit GLMs
fit &lt;- glmQLFit(carboplatin_counts, design)
carboplatin_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;carboplatin&#39;))
genes &lt;- topTags(carboplatin_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-bc6984d5c6eb5b2784f6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bc6984d5c6eb5b2784f6">{"x":{"filter":"none","vertical":false,"data":[["7002","6896","8582","17362","17785","4127","13818","10816","11952","10818"],["BTNL8","BNIP3","CCND2","MME","MXRA5","ACCN1","HAAO","DCAKD","EYA2","DCBLD2"],[0.08913661743224224,0.1120962483644857,0.1372016669157574,0.09443983206835704,-0.08238530711529678,0.08415046619951359,0.06764898485169879,0.05004348916152909,0.4661869402606638,-0.07875684058803736],[3.464405118216955,4.008417826634523,5.740253819220499,5.088266045527311,7.70149156013379,1.311144315980389,2.25127964787489,6.383041184740306,4.285079446053089,9.134881057099982],[93.11023029482618,99.13649114820758,78.67736705590218,64.49647610509894,109.1616288731054,49.24701030843642,50.61005179814405,71.14666680770104,102.0061756479931,58.47586625427644],[1.239683145984123e-06,1.479910488762893e-06,4.906287377226864e-06,7.892484378974147e-06,9.832755450279723e-06,2.151032886033691e-05,3.310731034317047e-05,4.369299854742175e-05,6.888980471213253e-05,8.503351667587755e-05],[0.01171571138429144,0.01171571138429144,0.02589374934787764,0.03113640340885577,0.03113640340885577,0.05676217280761905,0.0748840063804883,0.08647390575016606,0.1211924753341327,0.1220554596194387]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- carboplatin

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  carboplatin_counts$cpm &lt;- cpm(carboplatin_counts, log = T) 
  expression_values &lt;- carboplatin_counts$cpm[carboplatin_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-19-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="oxaliplatin" class="section level3">
<h3>Oxaliplatin</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 oxaliplatin
ORG-38-ORG           1      0      0       223.8
ORG41                1      0      0       225.3
ORG46                1      1      0       215.7
ORG49                1      1      0       498.9
ORG60                1      1      0       211.1
ORG66                1      0      1       193.5
ORG71                1      0      1       207.9
ORG73R               1      0      1       224.8</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(oxaliplatin_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>oxaliplatin_counts &lt;- oxaliplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
oxaliplatin_counts &lt;- calcNormFactors(oxaliplatin_counts)
#fit GLMs
fit &lt;- glmQLFit(oxaliplatin_counts, design)
oxaliplatin_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;oxaliplatin&#39;))
genes &lt;- topTags(oxaliplatin_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-56a7d65cc4ec04a0ed6b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-56a7d65cc4ec04a0ed6b">{"x":{"filter":"none","vertical":false,"data":[["6896","7002","8482","13402","11980","20957","20271","30432","34761","19251"],["BNIP3","BTNL8","CCDC43","GPATCH8","FABP1","RASL11A","PPYR1","SLC28A1","WFDC2","PARVB"],[0.03172884277058328,0.02259158043366917,0.01459919195101643,0.01544682996427507,0.06645555981023339,0.02122508528295213,0.0281304020952088,0.0216816905526389,-0.02182709002362028,0.01814025432099123],[4.008417826634523,3.464405118216955,6.164739044226242,8.408471247223426,4.09821646085354,3.82522843613651,2.877401604024115,1.663590540040651,7.786873850246892,3.306895650307708],[87.14072435858695,67.83965665314447,103.151830399288,101.1628376047618,150.9927291729877,83.04154451682714,41.99065363686982,44.34981775685836,65.51019328094178,65.17606019709594],[2.93429453947991e-06,5.570870037209881e-06,1.084649168972308e-05,1.158730376163996e-05,1.702504231252348e-05,2.250621101721879e-05,4.84946092986469e-05,5.17449263373635e-05,5.329071206259645e-05,5.480124079818758e-05],[0.04410179264957202,0.04410179264957202,0.04586544511451136,0.04586544511451136,0.05391149898683685,0.05939013983927085,0.07393702579860476,0.07393702579860476,0.07393702579860476,0.07393702579860476]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- oxaliplatin

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  oxaliplatin_counts$cpm &lt;- cpm(oxaliplatin_counts, log = T) 
  expression_values &lt;- oxaliplatin_counts$cpm[oxaliplatin_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-22-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cisplatin" class="section level3">
<h3>Cisplatin</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 cisplatin
ORG-38-ORG           1      0      0     205.1
ORG41                1      0      0     190.2
ORG46                1      1      0     187.1
ORG49                1      1      0     389.5
ORG60                1      1      0     195.2
ORG66                1      0      1     193.8
ORG71                1      0      1     200.5
ORG73R               1      0      1     220.2</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(cisplatin_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>cisplatin_counts &lt;- cisplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
cisplatin_counts &lt;- calcNormFactors(cisplatin_counts)
#fit GLMs
fit &lt;- glmQLFit(cisplatin_counts, design)
cisplatin_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;cisplatin&#39;))
genes &lt;- topTags(cisplatin_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-fe3aef495c8825682f3e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-fe3aef495c8825682f3e">{"x":{"filter":"none","vertical":false,"data":[["7002","6896","13402","8482","10788","20957","20640","11544","6155","30432"],["BTNL8","BNIP3","GPATCH8","CCDC43","DBF4B","RASL11A","PTPN13","EFTUD2","AQP1","SLC28A1"],[0.03318893564260637,0.0444835906713115,0.02227651428103004,0.0206859095445234,0.02466002017105618,0.03042263444350246,-0.02494418232369812,0.01673099734788423,0.02531014897269609,0.03132395615309172],[3.464405118216955,4.008417826634523,8.408471247223426,6.164739044226242,6.215785091334277,3.82522843613651,6.141934051361056,8.060506282847577,3.59669805887161,1.663590540040651],[82.97141875725544,74.51799148601931,119.81814316602,90.4231541520969,87.18723289387086,77.34385122369767,75.05771936496274,71.8511913262675,59.17593192518524,44.22685586695333],[1.89651169973676e-06,5.85748774144373e-06,6.505399450773216e-06,1.777186523544829e-05,2.029044061262109e-05,2.878680931411387e-05,3.403098860091849e-05,3.958900711071354e-05,4.171789073967464e-05,4.987211679882543e-05],[0.03002746974193212,0.03433332983469745,0.03433332983469745,0.06425170924392594,0.06425170924392594,0.07257635603693675,0.07257635603693675,0.07257635603693675,0.07257635603693675,0.07257635603693675]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- cisplatin

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  cisplatin_counts$cpm &lt;- cpm(cisplatin_counts, log = T) 
  expression_values &lt;- cisplatin_counts$cpm[cisplatin_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-25-10.png" width="672" style="display: block; margin: auto;" />
### Gemcitabine</p>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 gemcitabine
ORG-38-ORG           1      0      0      155.10
ORG41                1      0      0       76.88
ORG46                1      1      0       51.96
ORG49                1      1      0       35.48
ORG60                1      1      0      145.70
ORG66                1      0      1       80.76
ORG71                1      0      1       78.67
ORG73R               1      0      1       55.47</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(gemcitabine_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>gemcitabine_counts &lt;- gemcitabine_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
gemcitabine_counts &lt;- calcNormFactors(gemcitabine_counts)
#fit GLMs
fit &lt;- glmQLFit(gemcitabine_counts, design)
gemcitabine_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;gemcitabine&#39;))
genes &lt;- topTags(gemcitabine_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-db3be56eac6ca817053b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-db3be56eac6ca817053b">{"x":{"filter":"none","vertical":false,"data":[["8323","32272","5454","20954","16294","30176","14429","12551","31133","16011"],["CASP1","TGM2","ALDH1A2","RASIP1","LRRTM1","SHISA2","HS6ST2","FGF19","SNORD3A","LHFP"],[-0.05120308433267493,0.03564686235980889,0.0675018947432397,0.04020806129432053,0.06672598964018649,0.05576635911772616,0.04509335521060386,0.05033801862315947,-0.05156839308045928,0.04702721305868766],[1.375752092843454,9.302269696324471,3.347807687852173,1.939768502071103,2.893452902632583,2.942701693914341,1.461103117493721,5.210875081089467,5.279926279582194,0.6171440144338834],[30.77529370279212,45.45786773374435,29.44023997973705,32.73908161519574,26.7790427114976,27.58558662064886,21.24436105522466,31.1606569758686,29.66509105337505,19.34434888761253],[0.00014371878044987,0.00017270882804563,0.0002861371357199131,0.0002904586729068162,0.0004948889026229349,0.0005890418244014113,0.000591589203733031,0.0005974948806604508,0.000694128064378166,0.0006972551196536875],[0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096,0.9630399043299096]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- gemcitabine

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  gemcitabine_counts$cpm &lt;- cpm(gemcitabine_counts, log = T) 
  expression_values &lt;- gemcitabine_counts$cpm[gemcitabine_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-28-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="docetaxel" class="section level3">
<h3>Docetaxel</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 docetaxel
ORG-38-ORG           1      0      0    155.20
ORG41                1      0      0    101.20
ORG46                1      1      0     77.56
ORG49                1      1      0     63.52
ORG60                1      1      0    119.30
ORG66                1      0      1     68.85
ORG71                1      0      1    121.10
ORG73R               1      0      1     94.10</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(docetaxel_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>docetaxel_counts &lt;- docetaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
docetaxel_counts &lt;- calcNormFactors(docetaxel_counts)
#fit GLMs
fit &lt;- glmQLFit(docetaxel_counts, design)
docetaxel_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;docetaxel&#39;))
genes &lt;- topTags(docetaxel_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0c70ecafef5f4e0f6fd4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0c70ecafef5f4e0f6fd4">{"x":{"filter":"none","vertical":false,"data":[["33124","15652","7673","7922","16346","15535","6606","19422","15651","12158"],["TRPV6","KLK7","C20orf85","C6orf103","LY6D","KIF1A","ATRNL1","PDE3A","KLK6","FAM183A"],[-0.1117602404031037,-0.1340134624930241,0.1395988535384677,0.1132372350805323,-0.1131763717780774,0.1545935237154887,0.09590345420178784,0.1634699015303812,-0.07360157061385317,0.1079665869403383],[2.272419657042916,6.435273494070187,3.450442185996617,1.133927608669762,4.601543657220141,4.593245160778442,0.3832695284381713,5.170557871237515,6.301687524051877,1.67502615704263],[72.19048512103545,72.19708894847732,63.8925332966195,52.16007189342155,56.10162195328976,56.80405972721967,40.66110349396791,53.55887936924188,53.38493461059196,43.84121385802053],[9.601212348902036e-06,1.167045814948319e-05,2.731954042876868e-05,3.293168389898213e-05,4.988521523786054e-05,5.861665044701938e-05,7.06580193434459e-05,7.107300732807527e-05,8.959854152177399e-05,0.0001023504916889231],[0.09238918194038365,0.09238918194038365,0.130351837793146,0.130351837793146,0.140662365628177,0.140662365628177,0.140662365628177,0.140662365628177,0.1576237453238053,0.162051533491072]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- docetaxel

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  docetaxel_counts$cpm &lt;- cpm(docetaxel_counts, log = T) 
  expression_values &lt;- docetaxel_counts$cpm[docetaxel_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-31-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="paclitaxel" class="section level3">
<h3>Paclitaxel</h3>
<p>Design matrix:</p>
<pre class="r"><code>print(design)</code></pre>
<pre><code>           (Intercept) batch2 batch3 paclitaxel
ORG-38-ORG           1      0      0     157.30
ORG41                1      0      0     103.10
ORG46                1      1      0      69.78
ORG49                1      1      0      62.57
ORG60                1      1      0     113.60
ORG66                1      0      1      67.05
ORG71                1      0      1     122.00
ORG73R               1      0      1      91.94</code></pre>
<pre class="r"><code>#filtering to remove low counts
keep &lt;- filterByExpr(paclitaxel_counts, design = design)
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
22950 15833 </code></pre>
<pre class="r"><code>paclitaxel_counts &lt;- paclitaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
paclitaxel_counts &lt;- calcNormFactors(paclitaxel_counts)
#fit GLMs
fit &lt;- glmQLFit(paclitaxel_counts, design)
paclitaxel_auc_fit &lt;- glmQLFTest(fit, coef=c(&#39;paclitaxel&#39;))
genes &lt;- topTags(paclitaxel_auc_fit)

datatable(as.data.frame(genes))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ba594513a0e0a595a0ee" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ba594513a0e0a595a0ee">{"x":{"filter":"none","vertical":false,"data":[["15652","7673","7922","33124","6606","34633","12158","15535","15651","11136"],["KLK7","C20orf85","C6orf103","TRPV6","ATRNL1","VWA3A","FAM183A","KIF1A","KLK6","DNAI1"],[-0.1337581329744221,0.1439298822463358,0.1135199644612706,-0.1142957811276827,0.09660340608027201,0.09375614590757687,0.107847344226407,0.1528173579473741,-0.0743382858402509,0.1085776570668079],[6.435273494070187,3.450442185996617,1.133927608669762,2.272419657042916,0.3832695284381713,3.660329975186081,1.67502615704263,4.593245160778442,6.301687524051877,3.522985568772999],[109.8187680419585,96.89349809397473,54.87242902372496,54.15001223472085,45.47593858863441,65.38303575420596,45.45347570367201,52.07888371584792,51.24558064676035,58.24997235480245],[2.021394258560069e-06,5.181218905755521e-06,3.136053472215409e-05,4.268419653283291e-05,4.650950536469809e-05,7.471668273436258e-05,9.220322249573371e-05,0.0001037089958890063,0.000103867590546378,0.000105191622894457],[0.03200473529578157,0.04101711946741358,0.147276999687853,0.147276999687853,0.147276999687853,0.1499068940265682,0.1499068940265682,0.1499068940265682,0.1499068940265682,0.1499068940265682]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>F<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"genes","targets":1},{"name":"logFC","targets":2},{"name":"logCPM","targets":3},{"name":"F","targets":4},{"name":"PValue","targets":5},{"name":"FDR","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Plotting top genes:</p>
<pre class="r"><code>top_genes &lt;- c(genes$table$genes)
auc_values &lt;- paclitaxel

for (gene in top_genes) {
  gene &lt;- as.character(gene)
  paclitaxel_counts$cpm &lt;- cpm(paclitaxel_counts, log = T) 
  expression_values &lt;- paclitaxel_counts$cpm[paclitaxel_counts$genes$genes==gene,]
  plot &lt;- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab(&#39;AUC values&#39;) + 
    ylab(&#39;log(cpm)&#39;) +
    ggtitle(label=gene)
  print(plot)
}</code></pre>
<p><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Updated_DrugAUC.Rmd/unnamed-chunk-34-10.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] DT_0.33                org.Hs.eg.db_3.19.1    AnnotationDbi_1.67.0  
 [4] IRanges_2.39.2         S4Vectors_0.43.2       Biobase_2.65.0        
 [7] BiocGenerics_0.51.0    enrichplot_1.25.0      clusterProfiler_4.13.0
[10] RColorBrewer_1.1-3     ggpubr_0.6.0           edgeR_4.3.7           
[13] limma_3.61.7           ggplot2_3.5.1          pheatmap_1.0.12       
[16] workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] rstudioapi_0.16.0       jsonlite_1.8.8          magrittr_2.0.3         
  [4] farver_2.1.2            rmarkdown_2.27          fs_1.6.4               
  [7] zlibbioc_1.51.1         vctrs_0.6.5             memoise_2.0.1          
 [10] ggtree_3.13.0           rstatix_0.7.2           htmltools_0.5.8.1      
 [13] broom_1.0.6             gridGraphics_0.5-1      sass_0.4.9             
 [16] bslib_0.8.0             htmlwidgets_1.6.4       plyr_1.8.9             
 [19] cachem_1.1.0            whisker_0.4.1           igraph_2.0.3           
 [22] lifecycle_1.0.4         pkgconfig_2.0.3         gson_0.1.0             
 [25] Matrix_1.7-0            R6_2.5.1                fastmap_1.2.0          
 [28] GenomeInfoDbData_1.2.12 digest_0.6.36           aplot_0.2.3            
 [31] colorspace_2.1-1        patchwork_1.2.0         ps_1.7.7               
 [34] rprojroot_2.0.4         crosstalk_1.2.1         RSQLite_2.3.7          
 [37] labeling_0.4.3          fansi_1.0.6             httr_1.4.7             
 [40] polyclip_1.10-7         abind_1.4-5             compiler_4.4.0         
 [43] bit64_4.0.5             withr_3.0.1             backports_1.5.0        
 [46] BiocParallel_1.39.0     carData_3.0-5           viridis_0.6.5          
 [49] DBI_1.2.3               highr_0.11              ggforce_0.4.2          
 [52] ggsignif_0.6.4          MASS_7.3-61             HDO.db_0.99.1          
 [55] tools_4.4.0             scatterpie_0.2.3        ape_5.8                
 [58] httpuv_1.6.15           glue_1.7.0              callr_3.7.6            
 [61] nlme_3.1-165            GOSemSim_2.31.0         promises_1.3.0         
 [64] shadowtext_0.1.4        grid_4.4.0              getPass_0.2-4          
 [67] reshape2_1.4.4          fgsea_1.31.0            generics_0.1.3         
 [70] gtable_0.3.5            tidyr_1.3.1             data.table_1.15.4      
 [73] tidygraph_1.3.1         car_3.1-2               utf8_1.2.4             
 [76] XVector_0.45.0          ggrepel_0.9.5           pillar_1.9.0           
 [79] stringr_1.5.1           yulab.utils_0.1.5       later_1.3.2            
 [82] splines_4.4.0           dplyr_1.1.4             tweenr_2.0.3           
 [85] treeio_1.29.0           lattice_0.22-6          bit_4.0.5              
 [88] tidyselect_1.2.1        GO.db_3.19.1            locfit_1.5-9.10        
 [91] Biostrings_2.73.1       knitr_1.48              git2r_0.33.0           
 [94] gridExtra_2.3           xfun_0.46               graphlayouts_1.1.1     
 [97] statmod_1.5.0           stringi_1.8.4           UCSC.utils_1.1.0       
[100] lazyeval_0.2.2          ggfun_0.1.5             yaml_2.3.10            
[103] evaluate_0.24.0         codetools_0.2-20        ggraph_2.2.1           
[106] tibble_3.2.1            qvalue_2.37.0           ggplotify_0.1.2        
[109] cli_3.6.3               munsell_0.5.1           processx_3.8.4         
[112] jquerylib_0.1.4         Rcpp_1.0.13             GenomeInfoDb_1.41.1    
[115] png_0.1-8               parallel_4.4.0          blob_1.2.4             
[118] DOSE_3.31.2             tidytree_0.4.6          viridisLite_0.4.2      
[121] scales_1.3.0            purrr_1.0.2             crayon_1.5.3           
[124] rlang_1.1.4             cowplot_1.1.3           fastmatch_1.1-4        
[127] KEGGREST_1.45.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
