---
title: "Gene Set Analysis"
output: html_document
date: "2024-07-22"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
library(edgeR)
library(limma)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
library(ggpubr)
library(fgsea)
```
## Overview/Notes
Plots used for analysis: 

• Dot plot: number of genes belonging to each GO-term and associated adjusted p-values  

• Enrichment map: Shows overlap between genes in different GO terms

• Category Net plot:Linkages of genes and biological concepts as a network

• Ridge Plot: Density plots of expression for different gene sets to determine up/downregulated pathways 

```{r, include=FALSE}

#Loading data and creating design matrices for different gene sets
countdata <- read.delim("/Users/dalvinikita/Documents/RNA-seq Analysis/Sequencing Data/collated.count.tsv", check.names = FALSE, stringsAsFactors = FALSE)
sampleinfo <- read.delim("/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt", check.names=FALSE, stringsAsFactors = TRUE)
counts <- DGEList(counts=countdata[,c(7,10,14,17,20,24,25,28)], genes = countdata[,2])
batch <- sampleinfo$Batch

#TOP1 inhibitors
#calculating avg AUC for TOP1 inhibitors
y<- counts
top1_auc <- cbind(sampleinfo$Irinotecan,sampleinfo$Topotecan)
ave_auc <- rowMeans(top1_auc)
design <- model.matrix(~batch+ave_auc)
#filtering to remove low counts and fitting GLMs
keep <- filterByExpr(y, design = design)
y <- y[keep, , keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=TRUE)
fit <- glmFit(y, design)
lrt<- glmLRT(fit)
sig_genes <- topTags(lrt, p.value=0.05, n=500)
write.table(sig_genes, quote=F, row.names = F, col.names = T, file='/Users/dalvinikita/Documents/Drug Response Analysis/top1_drugs_genes.txt', sep='\t')

#Platinum based drugs
y<- counts
#calculating avg AUC for platinum based drugs
platinum_auc <- cbind(sampleinfo$Carboplatin,sampleinfo$Cisplatin)
platinum_auc <- cbind(platinum_auc,sampleinfo$Oxaliplatin)
ave_auc <- rowMeans(platinum_auc)
design <- model.matrix(~batch+ave_auc)
#filtering to remove low counts and fitting GLMs
keep <- filterByExpr(y, design = design)
y <- y[keep, , keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=TRUE)
fit <- glmFit(y, design)
lrt<- glmLRT(fit)
sig_genes <- topTags(lrt, p.value=0.05, n=500)
write.table(sig_genes, quote=F, row.names = F, col.names = T, file='/Users/dalvinikita/Documents/Drug Response Analysis/plat_drugs_genes.txt', sep='\t')

#Cytotoxic Antibiotics
y<- counts
#calculating avg AUC for cytotoxic antibiotics
cytotoxic_antibiotics <- cbind(sampleinfo$Mitomycin,sampleinfo$Doxorubicin)
ave_auc <- rowMeans(cytotoxic_antibiotics)
design <- model.matrix(~batch+ave_auc)
#filtering to remove low counts and fitting GLMs
keep <- filterByExpr(y, design = design)
y <- y[keep, , keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=TRUE)
fit <- glmFit(y, design)
lrt<- glmLRT(fit, coef=3)
sig_genes <- topTags(lrt, p.value=0.05, n=500)
write.table(sig_genes, quote=F, row.names = F, col.names = T, file='/Users/dalvinikita/Documents/Drug Response Analysis/cytotoxic_antibiotic_genes.txt', sep='\t')

#Taxanes
y<- counts
#calculating avg AUC for taxanes
taxanes <- cbind(sampleinfo$Paclitaxel,sampleinfo$Docetaxel)
ave_auc <- rowMeans(taxanes)
design <- model.matrix(~batch+ave_auc)
#filtering to remove low counts and fitting GLMs
keep <- filterByExpr(y, design = design)
y <- y[keep, , keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=TRUE)
fit <- glmFit(y, design)
lrt<- glmLRT(fit, coef=3)
sig_genes <- topTags(lrt, p.value=0.05, n=500)
write.table(sig_genes, quote=F, row.names = F, col.names = T, file='/Users/dalvinikita/Documents/Drug Response Analysis/taxane_genes.txt', sep='\t')

#Creating gsea objects
#TOP1
top1_sig_genes <- read.delim('/Users/dalvinikita/Documents/Drug Response Analysis/top1_drugs_genes.txt', header=T)
gene_list <- top1_sig_genes$logFC
names(gene_list)<- top1_sig_genes$genes
gene_list=sort(gene_list, decreasing = TRUE)

top1_gse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ALIAS", 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")

#Platinum compounds
plat_sig_genes <- read.delim('/Users/dalvinikita/Documents/Drug Response Analysis/plat_drugs_genes.txt', header=T)
head(plat_sig_genes)
gene_list <- plat_sig_genes$logFC
names(gene_list)<- plat_sig_genes$genes
gene_list=sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ALIAS", 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")

#Cytotoxic antibiotics
cytotoxic_antibiotic_genes <- read.delim('/Users/dalvinikita/Documents/Drug Response Analysis/cytotoxic_antibiotic_genes.txt', header=T)
head(cytotoxic_antibiotic_genes)
ca_gene_list <- cytotoxic_antibiotic_genes$logFC
names(ca_gene_list)<- cytotoxic_antibiotic_genes$genes
ca_gene_list=sort(ca_gene_list, decreasing = TRUE)

ca_gse <- gseGO(geneList=ca_gene_list, 
             ont ="ALL", 
             keyType = "ALIAS", 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")

#Taxanes
taxane_genes <- read.delim('/Users/dalvinikita/Documents/Drug Response Analysis/taxane_genes.txt', header=T)
head(taxane_genes)
tax_gene_list <- taxane_genes$logFC
names(tax_gene_list)<- taxane_genes$genes
tax_gene_list=sort(tax_gene_list, decreasing = TRUE)

tax_gse <- gseGO(geneList=tax_gene_list, 
             ont ="ALL", 
             keyType = "ALIAS", 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
```
## Dotplot {.tabset}

### TOP1 inhibitors
```{r, echo=FALSE}
require(DOSE)
dotplot(top1_gse, showCategory=8,split=".sign",label_format=50,color='p.adjust')+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle("dotplot for TOP1 inhibitors")
```

### Platinum compounds
```{r, echo=FALSE}
require(DOSE)
dotplot(gse, showCategory=8, split=".sign",label_format=50,color='p.adjust')+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle("dotplot for platinum based compounds")
```

### Cytotoxic Antibiotics
```{r, echo=FALSE}
require(DOSE)
dotplot(ca_gse, showCategory=8, split=".sign",label_format=50,color='p.adjust')+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle("dotplot for cytotoxic antibiotics")
```

### Taxanes
```{r, echo=FALSE}
require(DOSE)
dotplot(tax_gse, showCategory=8, split=".sign",label_format=50,color='p.adjust')+
  facet_grid(.~.sign) +
  theme_dose(font.size = 5)+
  ggtitle("dotplot for taxanes")
```

## {-}
Comments: Gene sets for different drug groups are generally quite different with minimal overlap. Genes identified for TOP1 inhibitors seem to be involved in cell cycle processes and ion channel activity. Pathways altered in chemotherapy resistance for platinum based compounds relate to regulating DNA/RNA binding activity and ion transport.Catabolic and cellular processes and pathways relating to the golgi lumen and vesicle activity were identified for cytotoxic antibiotics. Taxane resistance was associated with altered expression of genes related to molecular function and filament processes.  

## Enrichment Map {.tabset}

### TOP1 inhibitors
```{r, echo=FALSE}
x<- pairwise_termsim(top1_gse)
emapplot(x,layout.params=list(layout='circle'))+
  ggtitle("enrichment map for TOP1 inhibitors")
```

### Platinum compounds
```{r, echo=FALSE}
x<- pairwise_termsim(gse)
emapplot(x, showCategory=20,layout.params=list(layout='circle'))+
  ggtitle("enrichment map for platinum based compounds")
```

### Cytotoxic Antibiotics
```{r, echo=FALSE}
x<- pairwise_termsim(ca_gse)
emapplot(x, showCategory=40)+
  ggtitle("enrichment map for cytotoxic antibiotics")
```

### Taxanes
```{r, echo=FALSE}
x<- pairwise_termsim(tax_gse)
emapplot(x, showCategory=40)+
  ggtitle("enrichment map for taxanes")
```

## Category netplot {.tabset}

### TOP1 inhibitors
```{r, echo=FALSE}

cnetplot(top1_gse, node_label="category", showCategory=10, color.params=list(category='deeppink1', gene='darkmagenta'))+
  ggtitle("By gene category")

cnetplot(top1_gse, node_label="gene", color.params=list(category='deeppink1', gene='darkmagenta'), showCategory=10)+
  ggtitle("By gene name")
```

### Platinum compounds
```{r, echo=FALSE}
cnetplot(gse, node_label="category", showCategory=10, color.params=list(category='deeppink1', gene='darkmagenta'))+
  ggtitle("By gene category")

cnetplot(gse, node_label="gene", color.params=list(category='deeppink1', gene='darkmagenta'), showCategory=10)+
  ggtitle("By gene name")
```

### Cytotoxic Antibiotics
```{r, message=FALSE, warning=FALSE, echo=FALSE}

cnetplot(ca_gse, node_label="category", showCategory=10, color.params=list(category='deeppink1', gene='darkmagenta'))+
  ggtitle("By gene category")

cnetplot(ca_gse, node_label="gene", color.params=list(category='deeppink1', gene='darkmagenta'), showCategory=10)+
  ggtitle("By gene name")

```

### Taxanes
```{r, warning=FALSE, echo=FALSE}

cnetplot(tax_gse, node_label="category", showCategory=10, color.params=list(category='deeppink1', gene='darkmagenta'))+
  ggtitle("By gene category")

cnetplot(tax_gse, node_label="gene", color.params=list(category='deeppink1', gene='darkmagenta'), showCategory=10)+
  ggtitle("By gene name")
```

## Ridgeplot {.tabset}

### TOP1 inhibitors
```{r, echo=FALSE}
ridgeplot(top1_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle("ridgeplot for TOP1 inhibitors")
```

### Platinum compounds
```{r, echo=FALSE}
ridgeplot(gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle("ridgeplot for platinum based compounds")
```

### Cytotoxic Antibiotics
```{r, message=FALSE, warning=FALSE, echo=FALSE}
ridgeplot(ca_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle("ridgeplot for cytotoxic antibiotics")
```

### Taxanes
```{r, warning=FALSE, echo=FALSE}
ridgeplot(tax_gse, showCategory = 15, label_format =50 )+theme_dose(font.size = 5)+
  ggtitle("ridgeplot for taxanes")
```

## {-}
Comments: logFC shown in ridge plots are all significant (p<0.05) for all drugs except cytotoxic antibiotics (of which no pathway shows statistical significance). 

