---
title: "Updated Drug AUC correlations"
output: html_document
date: "2024-08-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = FALSE,
	message = FALSE,
	warning = FALSE
)
library(pheatmap)
library(ggplot2)
library(edgeR)
library(limma)
library(ggpubr)
library(RColorBrewer)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(DT)

```

## Differential Expression of Organoid VS Tumour samples 
A quick look at the data: 
```{r, echo=TRUE}

# loading in data
rawdata <- read.delim2('/Users/dalvinikita/Documents/RNA-seq Analysis/Sequencing Data/collated.count.tsv', header = TRUE, check.names = FALSE, stringsAsFactors = FALSE)

head(rawdata) 

sampleinfo <- read.delim2('/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt', header=TRUE, check.names = FALSE, stringsAsFactors = TRUE)

head(sampleinfo)

```

### QC and Normalisation
```{r, fig.height=10, fig.width=30}

#creating a DGElist object 
y<- DGEList(counts=rawdata[,c(7,10,14,17,20,24,25,28)], genes=rawdata[,2])

#create factors to include in design matrix
sampleinfo$Batch<- as.factor(sampleinfo$Batch)
batch <- (c(sampleinfo$Batch))
patient <- as.factor(c(sampleinfo$PATIENT))
pattern <- as.factor(c(sampleinfo$PATTERN))

#create design matrix to include patient, batch, growth pattern
design <- model.matrix(~patient+pattern+batch)
rownames(design) <- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y, design)
table(keep)
y <- y[keep, , keep.lib.sizes=TRUE]

#Visualiszing library sizes pre and post normalisation
col <- brewer.pal(32, 'Paired')

par(mfrow=c(1,2))
lcpm <- cpm(y, log=TRUE)
boxplot(lcpm, las=2, col=col, main="")
title(main="A. Example: Unnormalised data",ylab="Log-cpm")

y <- calcNormFactors(y) 
lcpm <- cpm(y, log=TRUE)
boxplot(lcpm, col=col, main="")
title(main="B. Example: Normalised data",ylab="Log-cpm")
```
### MDS plots 
```{r, fig.height=7, fig.width=14}
par(mfrow=c(1,2))

col.pattern <- c('darkorchid3', 'deeppink','darkolivegreen1')[sampleinfo$PATTERN]
col.batch <- c('darkorchid3', 'deeppink','darkolivegreen1')[sampleinfo$Batch]


plotMDS(y, col=col.pattern, pch=19)
title(main="B. Growth Pattern")
legend("topleft",
       fill=c('darkorchid3', 'deeppink','darkolivegreen1'), 
       legend=levels(sampleinfo$PATTERN))

plotMDS(y, col=col.batch, pch=19)
title(main="A. Batch")
legend("topleft",
       fill=c('darkorchid3', 'deeppink','darkolivegreen1'), 
       legend=levels(sampleinfo$Batch))

```

### Fitting a Linear Model {.tabset}
```{r}
#reset DGE object
sampleinfo <- read.delim2('/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt', header=TRUE, check.names = FALSE, stringsAsFactors = FALSE)
y<- DGEList(counts=rawdata[,c(7,10,14,17,20,24,25,28)], genes=rawdata[,2])
batch <- as.factor(sampleinfo$Batch)
```


#### Mitomycin 
```{r echo=FALSE, fig.height=30, fig.width=20}
mito_counts <- y
design <- model.matrix(~batch+sampleinfo$Mitomycin)
mitomycin <- as.numeric(sampleinfo$Mitomycin)
design <- cbind(design[,1:3],mitomycin)
#filtering to remove low counts
keep <- filterByExpr(mito_counts, design = design)
table(keep)
mito_counts <- mito_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
mito_counts <- calcNormFactors(mito_counts)
#fit GLMs
fit <- glmQLFit(mito_counts, design)
mito_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(mito_auc_fit)

datatable(as.data.frame(genes))

```

#### Irinotecan
```{r echo=FALSE, fig.height=30, fig.width=20}
irino_counts <- y
design <- model.matrix(~batch+sampleinfo$Irinotecan)
irinotecan <- as.numeric(sampleinfo$Irinotecan)
design<- cbind(design[,1:3],irinotecan)

#filtering to remove low counts
keep <- filterByExpr(irino_counts, design = design)
table(keep)
irino_counts <- irino_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
irino_counts <- calcNormFactors(irino_counts)
#fit GLMs
fit <- glmQLFit(irino_counts, design)
irinotecan_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(irinotecan_auc_fit)

datatable(as.data.frame(genes))
```

#### Topotecan
```{r echo=FALSE, fig.height=30, fig.width=20}
topotecan_counts <- y
design <- model.matrix(~batch+sampleinfo$Topotecan)
topotecan <- as.numeric(sampleinfo$Topotecan)
design<- cbind(design[,1:3],topotecan)

#filtering to remove low counts
keep <- filterByExpr(topotecan_counts, design = design)
table(keep)
topotecan_counts <- topotecan_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
topotecan_counts <- calcNormFactors(topotecan_counts)
#fit GLMs
fit <- glmQLFit(topotecan_counts, design)
topotecan_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(topotecan_auc_fit)

datatable(as.data.frame(genes))

```

#### Doxorubicin
```{r echo=FALSE, fig.height=30, fig.width=20}
doxorubicin_counts <- y
design <- model.matrix(~batch+sampleinfo$Doxorubicin)
doxorubicin <- as.numeric(sampleinfo$Doxorubicin)
design<- cbind(design[,1:3],doxorubicin)

#filtering to remove low counts
keep <- filterByExpr(doxorubicin_counts, design = design)
table(keep)
doxorubicin_counts <- doxorubicin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
doxorubicin_counts <- calcNormFactors(doxorubicin_counts)
#fit GLMs
fit <- glmQLFit(doxorubicin_counts, design)
doxorubicin_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(topotecan_auc_fit)

datatable(as.data.frame(genes))
```

#### Carboplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
carboplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Carboplatin)
carboplatin <- as.numeric(sampleinfo$Carboplatin)
design<- cbind(design[,1:3],carboplatin)

#filtering to remove low counts
keep <- filterByExpr(carboplatin_counts, design = design)
table(keep)
carboplatin_counts <- carboplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
carboplatin_counts <- calcNormFactors(carboplatin_counts)
#fit GLMs
fit <- glmQLFit(carboplatin_counts, design)
carboplatin_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(carboplatin_auc_fit)

datatable(as.data.frame(genes))
```

#### Oxaliplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
oxaliplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Oxaliplatin)
oxaliplatin <- as.numeric(sampleinfo$Oxaliplatin)
design<- cbind(design[,1:3],oxaliplatin)

#filtering to remove low counts
keep <- filterByExpr(oxaliplatin_counts, design = design)
table(keep)
oxaliplatin_counts <- oxaliplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
oxaliplatin_counts <- calcNormFactors(oxaliplatin_counts)
#fit GLMs
fit <- glmQLFit(oxaliplatin_counts, design)
oxaliplatin_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(oxaliplatin_auc_fit)

datatable(as.data.frame(genes))
```

#### Cisplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
cisplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Cisplatin)
cisplatin <- as.numeric(sampleinfo$Cisplatin)
design<- cbind(design[,1:3],cisplatin)

#filtering to remove low counts
keep <- filterByExpr(cisplatin_counts, design = design)
table(keep)
cisplatin_counts <- cisplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
cisplatin_counts <- calcNormFactors(cisplatin_counts)
#fit GLMs
fit <- glmQLFit(cisplatin_counts, design)
cisplatin_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(cisplatin_auc_fit)

datatable(as.data.frame(genes))
```

#### Gemcitabine
```{r echo=FALSE, fig.height=30, fig.width=20}
gemcitabine_counts <- y
design <- model.matrix(~batch+sampleinfo$Gemcitabine)
gemcitabine <- as.numeric(sampleinfo$Gemcitabine)
design<- cbind(design[,1:3],gemcitabine)

#filtering to remove low counts
keep <- filterByExpr(gemcitabine_counts, design = design)
table(keep)
gemcitabine_counts <- gemcitabine_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
gemcitabine_counts <- calcNormFactors(gemcitabine_counts)
#fit GLMs
fit <- glmQLFit(gemcitabine_counts, design)
gemcitabine_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(gemcitabine_auc_fit)

datatable(as.data.frame(genes))
```

#### Docetaxel
```{r echo=FALSE, fig.height=30, fig.width=20}
docetaxel_counts <- y
design <- model.matrix(~batch+sampleinfo$Docetaxel)
docetaxel <- as.numeric(sampleinfo$Docetaxel)
design<- cbind(design[,1:3],docetaxel)

#filtering to remove low counts
keep <- filterByExpr(docetaxel_counts, design = design)
table(keep)
docetaxel_counts <- docetaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
docetaxel_counts <- calcNormFactors(docetaxel_counts)
#fit GLMs
fit <- glmQLFit(docetaxel_counts, design)
docetaxel_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(docetaxel_auc_fit)

datatable(as.data.frame(genes))
```

#### Paclitaxel
```{r echo=FALSE, fig.height=30, fig.width=20}
paclitaxel_counts <- y
design <- model.matrix(~batch+sampleinfo$Paclitaxel)
paclitaxel <- as.numeric(sampleinfo$Paclitaxel)
design<- cbind(design[,1:3],paclitaxel)

#filtering to remove low counts
keep <- filterByExpr(paclitaxel_counts, design = design)
table(keep)
paclitaxel_counts <- paclitaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
paclitaxel_counts <- calcNormFactors(paclitaxel_counts)
#fit GLMs
fit <- glmQLFit(paclitaxel_counts, design)
paclitaxel_auc_fit <- glmLRT(fit, coef=4)
genes <- topTags(paclitaxel_auc_fit)

datatable(as.data.frame(genes))
```

