---
title: "Drug AUC Model"
output: html_document
date: "2024-08-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	fig.width = 5,
	fig.height=5
)
library(pheatmap)
library(ggplot2)
library(edgeR)
library(limma)
library(ggpubr)
library(RColorBrewer)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(DT)
library(EnhancedVolcano)
```

## RNA-seq data 
A quick look at the data: 
```{r}

# loading in data
rawdata <- read.delim2('/Volumes/bioinf/home/ndalvi/RNAseq_STAR/Results/HTseq/collated_counts.tsv', header = TRUE, check.names = FALSE, stringsAsFactors = FALSE)
 
head(rawdata) 

sampleinfo <- read.delim2('/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt', header=TRUE, check.names = FALSE, stringsAsFactors = TRUE)

head(sampleinfo)

```

### QC and Normalisation
```{r, fig.height=10, fig.width=30, echo=FALSE}

#creating a DGElist object 
y<- DGEList(counts=rawdata[,c("ORG-38-ORG","ORG41","ORG-46-ORG","ORG49","ORG55","ORG60","ORG61","ORG64_new","ORG65","ORG66","ORG70","ORG71","ORG73")], genes=rawdata[,1])

#create factors to include in design matrix
sampleinfo$Batch<- as.factor(sampleinfo$Batch)
batch <- (c(sampleinfo$Batch))
patient <- as.factor(c(sampleinfo$PATIENT))
pattern <- as.factor(c(sampleinfo$PATTERN))

#create design matrix to include patient, batch, growth pattern
design <- model.matrix(~0+patient+pattern+batch)
rownames(design) <- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y, design)
table(keep)
y <- y[keep, , keep.lib.sizes=TRUE]

#Visualiszing library sizes pre and post normalisation
col <- brewer.pal(32, 'Paired')

par(mfrow=c(1,2))
lcpm <- cpm(y, log=TRUE)
boxplot(lcpm, las=2, col=col, main="")
title(main="A. Example: Unnormalised data",ylab="Log-cpm")

y <- calcNormFactors(y) 
lcpm <- cpm(y, log=TRUE)
boxplot(lcpm, col=col, main="")
title(main="B. Example: Normalised data",ylab="Log-cpm")
```
### MDS plots 
```{r, fig.height=7, fig.width=14, echo=FALSE}
par(mfrow=c(1,2))

col.pattern <- c('darkorchid3', 'deeppink','darkolivegreen1')[sampleinfo$PATTERN]
col.batch <- c('darkorchid3', 'deeppink','darkolivegreen1')[sampleinfo$Batch]


plotMDS(y, col=col.pattern, pch=19)
title(main="B. Growth Pattern")
legend("topleft",
       fill=c('darkorchid3', 'deeppink','darkolivegreen1'), 
       legend=levels(sampleinfo$PATTERN))

plotMDS(y, col=col.batch, pch=19)
title(main="A. Batch")
legend("topleft",
       fill=c('darkorchid3', 'deeppink','darkolivegreen1'), 
       legend=levels(sampleinfo$Batch))

```

## Method 1: Fitting a GLM {.tabset}
```{r}
sampleinfo <- read.delim2('/Users/dalvinikita/Documents/Drug Response Analysis/updated_sampleInfo.txt', header=TRUE, check.names = FALSE, stringsAsFactors = FALSE)

y<- DGEList(counts=rawdata[,c("ORG-38-ORG","ORG41","ORG-46-ORG","ORG49","ORG55","ORG60","ORG61","ORG64_new","ORG65","ORG66","ORG70","ORG71","ORG73")], genes=rawdata[,1])

batch <- as.factor(sampleinfo$Batch)
```

### Mitomycin 
```{r, fig.height=30, fig.width=20}
mito_counts <- y
design <- model.matrix(~batch+sampleinfo$Mitomycin)
mitomycin <- as.numeric(sampleinfo$Mitomycin)
design <- cbind(design[,1:3],mitomycin)
rownames(design)<- sampleinfo$PATIENT
```

Design matrix:

```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(mito_counts, design = design)
table(keep)
mito_counts <- mito_counts[keep, , keep.lib.sizes=FALSE]

#normalizing library sizes
mito_counts <- calcNormFactors(mito_counts)

#fit GLMs
fit <- glmQLFit(mito_counts, design)
mito_auc_fit <- glmQLFTest(fit, coef=c('mitomycin'))
genes <- topTags(mito_auc_fit, n=Inf)

datatable(as.data.frame(genes))

EnhancedVolcano(genes$table,
                lab = rownames(genes), 
                x = 'logFC',
                y = 'FDR')
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- mitomycin

for (gene in top_genes) {
  gene <- as.character(gene)
  mito_counts$cpm <- cpm(mito_counts, log = T) 
  expression_values <- mito_counts$cpm[mito_counts$genes$genes==gene,]
  plot<- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}

```


### Irinotecan
```{r echo=FALSE, fig.height=30, fig.width=20}
irino_counts <- y
design <- model.matrix(~batch+sampleinfo$Irinotecan)
irinotecan <- as.numeric(sampleinfo$Irinotecan)
design<- cbind(design[,1:3],irinotecan)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(irino_counts, design = design)
table(keep)
irino_counts <- irino_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
irino_counts <- calcNormFactors(irino_counts)
#fit GLMs
fit <- glmQLFit(irino_counts, design)
irinotecan_auc_fit <- glmQLFTest(fit, coef=c('irinotecan'))
genes <- topTags(irinotecan_auc_fit, n=Inf)

EnhancedVolcano(genes$table,
                lab = rownames(genes), 
                x = 'logFC',
                y = 'FDR')

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- irinotecan

for (gene in top_genes) {
  gene <- as.character(gene)
  irino_counts$cpm <- cpm(irino_counts, log = T) 
  expression_values <- irino_counts$cpm[irino_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Topotecan
```{r echo=FALSE, fig.height=30, fig.width=20}
topotecan_counts <- y
design <- model.matrix(~batch+sampleinfo$Topotecan)
topotecan <- as.numeric(sampleinfo$Topotecan)
design<- cbind(design[,1:3],topotecan)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(topotecan_counts, design = design)
table(keep)
topotecan_counts <- topotecan_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
topotecan_counts <- calcNormFactors(topotecan_counts)
#fit GLMs
fit <- glmQLFit(topotecan_counts, design)
topotecan_auc_fit <- glmQLFTest(fit, coef=c('topotecan'))
genes <- topTags(topotecan_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- topotecan

for (gene in top_genes) {
  gene <- as.character(gene)
  topotecan_counts$cpm <- cpm(topotecan_counts, log = T) 
  expression_values <- topotecan_counts$cpm[topotecan_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Doxorubicin
```{r echo=FALSE, fig.height=30, fig.width=20}
doxorubicin_counts <- y
design <- model.matrix(~batch+sampleinfo$Doxorubicin)
doxorubicin <- as.numeric(sampleinfo$Doxorubicin)
design<- cbind(design[,1:3],doxorubicin)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(doxorubicin_counts, design = design)
table(keep)
doxorubicin_counts <- doxorubicin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
doxorubicin_counts <- calcNormFactors(doxorubicin_counts)
#fit GLMs
fit <- glmQLFit(doxorubicin_counts, design)
doxorubicin_auc_fit <- glmQLFTest(fit, coef=c('doxorubicin'))
genes <- topTags(doxorubicin_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- doxorubicin

for (gene in top_genes) {
  gene <- as.character(gene)
  doxorubicin_counts$cpm <- cpm(doxorubicin_counts, log = T) 
  expression_values <- doxorubicin_counts$cpm[doxorubicin_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Carboplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
carboplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Carboplatin)
carboplatin <- as.numeric(sampleinfo$Carboplatin)
design<- cbind(design[,1:3],carboplatin)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(carboplatin_counts, design = design)
table(keep)
carboplatin_counts <- carboplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
carboplatin_counts <- calcNormFactors(carboplatin_counts)
#fit GLMs
fit <- glmQLFit(carboplatin_counts, design)
carboplatin_auc_fit <- glmQLFTest(fit, coef=c('carboplatin'))
genes <- topTags(carboplatin_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- carboplatin

for (gene in top_genes) {
  gene <- as.character(gene)
  carboplatin_counts$cpm <- cpm(carboplatin_counts, log = T) 
  expression_values <- carboplatin_counts$cpm[carboplatin_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Oxaliplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
oxaliplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Oxaliplatin)
oxaliplatin <- as.numeric(sampleinfo$Oxaliplatin)
design<- cbind(design[,1:3],oxaliplatin)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(oxaliplatin_counts, design = design)
table(keep)
oxaliplatin_counts <- oxaliplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
oxaliplatin_counts <- calcNormFactors(oxaliplatin_counts)
#fit GLMs
fit <- glmQLFit(oxaliplatin_counts, design)
oxaliplatin_auc_fit <- glmQLFTest(fit, coef=c('oxaliplatin'))
genes <- topTags(oxaliplatin_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- oxaliplatin

for (gene in top_genes) {
  gene <- as.character(gene)
  oxaliplatin_counts$cpm <- cpm(oxaliplatin_counts, log = T) 
  expression_values <- oxaliplatin_counts$cpm[oxaliplatin_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Cisplatin
```{r echo=FALSE, fig.height=30, fig.width=20}
cisplatin_counts <- y
design <- model.matrix(~batch+sampleinfo$Cisplatin)
cisplatin <- as.numeric(sampleinfo$Cisplatin)
design<- cbind(design[,1:3],cisplatin)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(cisplatin_counts, design = design)
table(keep)
cisplatin_counts <- cisplatin_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
cisplatin_counts <- calcNormFactors(cisplatin_counts)
#fit GLMs
fit <- glmQLFit(cisplatin_counts, design)
cisplatin_auc_fit <- glmQLFTest(fit, coef=c('cisplatin'))
genes <- topTags(cisplatin_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- cisplatin

for (gene in top_genes) {
  gene <- as.character(gene)
  cisplatin_counts$cpm <- cpm(cisplatin_counts, log = T) 
  expression_values <- cisplatin_counts$cpm[cisplatin_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```
### Gemcitabine
```{r echo=FALSE, fig.height=30, fig.width=20}
gemcitabine_counts <- y
design <- model.matrix(~batch+sampleinfo$Gemcitabine)
gemcitabine <- as.numeric(sampleinfo$Gemcitabine)
design<- cbind(design[,1:3],gemcitabine)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(gemcitabine_counts, design = design)
table(keep)
gemcitabine_counts <- gemcitabine_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
gemcitabine_counts <- calcNormFactors(gemcitabine_counts)
#fit GLMs
fit <- glmQLFit(gemcitabine_counts, design)
gemcitabine_auc_fit <- glmQLFTest(fit, coef=c('gemcitabine'))
genes <- topTags(gemcitabine_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- gemcitabine

for (gene in top_genes) {
  gene <- as.character(gene)
  gemcitabine_counts$cpm <- cpm(gemcitabine_counts, log = T) 
  expression_values <- gemcitabine_counts$cpm[gemcitabine_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Docetaxel
```{r echo=FALSE, fig.height=30, fig.width=20}
docetaxel_counts <- y
design <- model.matrix(~batch+sampleinfo$Docetaxel)
docetaxel <- as.numeric(sampleinfo$Docetaxel)
design<- cbind(design[,1:3],docetaxel)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(docetaxel_counts, design = design)
table(keep)
docetaxel_counts <- docetaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
docetaxel_counts <- calcNormFactors(docetaxel_counts)
#fit GLMs
fit <- glmQLFit(docetaxel_counts, design)
docetaxel_auc_fit <- glmQLFTest(fit, coef=c('docetaxel'))
genes <- topTags(docetaxel_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- docetaxel

for (gene in top_genes) {
  gene <- as.character(gene)
  docetaxel_counts$cpm <- cpm(docetaxel_counts, log = T) 
  expression_values <- docetaxel_counts$cpm[docetaxel_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

### Paclitaxel
```{r echo=FALSE, fig.height=30, fig.width=20}
paclitaxel_counts <- y
design <- model.matrix(~batch+sampleinfo$Paclitaxel)
paclitaxel <- as.numeric(sampleinfo$Paclitaxel)
design<- cbind(design[,1:3],paclitaxel)
rownames(design)<- sampleinfo$`Sample ID`
```

Design matrix:
```{r}
print(design)

#filtering to remove low counts
keep <- filterByExpr(paclitaxel_counts, design = design)
table(keep)
paclitaxel_counts <- paclitaxel_counts[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
paclitaxel_counts <- calcNormFactors(paclitaxel_counts)
#fit GLMs
fit <- glmQLFit(paclitaxel_counts, design)
paclitaxel_auc_fit <- glmQLFTest(fit, coef=c('paclitaxel'))
genes <- topTags(paclitaxel_auc_fit)

datatable(as.data.frame(genes))
```

Plotting top genes: 
```{r}
top_genes <- c(genes$table$genes)
top_genes <- top_genes[1:10]
auc_values <- paclitaxel

for (gene in top_genes) {
  gene <- as.character(gene)
  paclitaxel_counts$cpm <- cpm(paclitaxel_counts, log = T) 
  expression_values <- paclitaxel_counts$cpm[paclitaxel_counts$genes$genes==gene,]
  plot <- ggplot(data = NULL, mapping = aes(x=auc_values, y=expression_values, colour = batch)) + 
    geom_point() +
    theme_linedraw() +
    xlab('AUC values') + 
    ylab('log(cpm)') +
    ggtitle(label=gene)
  print(plot)
}
```

## Method 2: Differential Expression Analysis {.tabset}
### TOP1 inhibitors
```{r}
y1 <- y
design <- model.matrix(~batch+sampleinfo$top_resistance)
rownames(design)<- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y1, design = design)
table(keep)
y1 <- y1[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
y1 <- calcNormFactors(y1)
y1 <- estimateDisp(y1, design, robust=TRUE)
fit <- glmQLFit(y1, design)
lrt <- glmQLFTest(fit, coef=3)
genes <- topTags(lrt, n=Inf)

EnhancedVolcano(genes$table,
                lab = genes$table$genes, 
                x = 'logFC',
                y = 'FDR', 
                pCutoff = 0.05,
                boxedLabels = TRUE,
                labSize = 4) + ggtitle("TOP1 inhibitors")
```

### Cytotoxic Antibiotics
```{r}
y2 <- y
design <- model.matrix(~batch+sampleinfo$ca_resistance)
rownames(design)<- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y2, design = design)
table(keep)
y2 <- y2[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
y2 <- calcNormFactors(y2)
y2 <- estimateDisp(y2, design, robust=TRUE)
fit <- glmQLFit(y2, design)
lrt <- glmQLFTest(fit, coef=3)
genes <- topTags(lrt, n=Inf)


EnhancedVolcano(genes$table,
                lab = genes$table$genes, 
                x = 'logFC',
                y = 'FDR', 
                pCutoff = 0.05,
                boxedLabels = TRUE,
                labSize = 4) + ggtitle("Cytotoxic Antibiotics")
```

### Taxanes 
```{r}
y3 <- y
design <- model.matrix(~batch+sampleinfo$taxane_resistance)
rownames(design)<- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y3, design = design)
table(keep)
y3 <- y3[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
y3 <- calcNormFactors(y3)
y3 <- estimateDisp(y3, design, robust=TRUE)
fit <- glmQLFit(y3, design)
lrt <- glmQLFTest(fit, coef=3)
genes <- topTags(lrt, n=Inf)

EnhancedVolcano(genes$table,
                lab = genes$table$genes, 
                x = 'logFC',
                y = 'FDR', 
                pCutoff = 0.05,
                boxedLabels = TRUE,
                labSize = 4)+ ggtitle("Taxanes")
```

### Platinum based
```{r, fig.height=7, fig.width=7}
y4 <- y
design <- model.matrix(~batch+sampleinfo$plat_resistance)
rownames(design)<- sampleinfo$`Sample ID`

#filtering to remove low counts
keep <- filterByExpr(y4, design = design)
table(keep)
y4 <- y4[keep, , keep.lib.sizes=FALSE]
#normalizing library sizes
y4 <- calcNormFactors(y4)
y4 <- estimateDisp(y4, design, robust=TRUE)
fit <- glmQLFit(y4, design)
lrt <- glmQLFTest(fit, coef=3)
genes <- topTags(lrt, n=Inf)

EnhancedVolcano(genes$table,
                lab = genes$table$genes, 
                x = 'logFC',
                y = 'FDR', 
                pCutoff = 0.05,
                boxedLabels = TRUE,
                labSize = 4)+ ggtitle("Platinum-based")
```
